browser.go

```diff
@@ -82,11 +82,18 @@ func (p *UserAgent) detectBrowser(sections []section) {
 				case "OPR":
 					p.browser.Name = "Opera"
 					p.browser.Version = sections[slen-1].version
+				case "Mobile":
+					p.browser.Name = "Mobile App"
```

Comment:
Carrying this comment over from #83:

@mssola I'm okay with calling this something different if you have any suggestions. This handles the case of embedded webviews in an iOS app that use UA strings like:

> Mozilla/5.0 (iPhone; CPU iPhone OS 14_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 [LinkedInApp]

[This website](https://user-agents.net/string/mozilla-5-0-iphone-cpu-iphone-os-14-7-like-mac-os-x-applewebkit-605-1-15-khtml-like-gecko-mobile-15e148-linkedinapp) calls the browser "LinkedInApp" while [this website](https://developers.whatismybrowser.com/useragents/parse/1240908webkit-based-browser-ios-iphone-webkit) calls it "WebKit based browser".

Those UA strings are currently being parsed as `Name = "Safari"` and `Version = "15E148"`, I mostly want to prevent an invalid Safari/Version combination from being returned.